lib/vehicle_factory.rb

```diff
@@ -0,0 +1,20 @@
+class VehicleFactory
+  attr_reader :factory_vehicles
+  def initialize
+    @factory_vehicles = []
+  end
+
+  def create_vehicles(state_registrations_data)
+    state_registrations_data.each do |vehicle|
+      # For now, since only washington, all are EV, so :ev is hard coded
+      vehicle_details = {
+        vin: vehicle[:vin_1_10], 
+        year: vehicle[:model_year], 
+        make: vehicle[:make], 
+        model: vehicle[:model], 
+        engine: :ev
+      }
+      @factory_vehicles << Vehicle.new(vehicle_details)
+    end
+  end
+end
```

Comment:
In our Interaction Pattern, I don't believe it mentions that we need to be able to call `factory_vehicles` on a `VehicleFactory` object. For that reason, I don't think we need an attr_reader for that attribute. I would even go a bit further to say you don't need any attribute to get set upon initialization. That's because your `create_vehicles` method is meant to iterate and calculate those vehicles for you. We could refactor this class to look something like: 


```ruby 
class VehicleFactory

  def create_vehicles(state_registrations_data)
    state_registrations_data.map do |vehicle|
      vehicle_details = {
        vin: vehicle[:vin_1_10], 
        year: vehicle[:model_year], 
        make: vehicle[:make], 
        model: vehicle[:model], 
        engine: :ev
      }
    end
  end
end
```